site/en/docs/web-platform/bfcache-notrestoredreasons/index.md

```diff
@@ -0,0 +1,216 @@
+
+```js
+{
+  blocked: false,
+  children: [
+    {url: "", src: "b.com", id: "b", name: "b", blocked: null, reasons: [], children: []},
+    {url: "", src: "c.com", id: "c", name: "c", blocked: true, reasons: [], children: []},
+    {url: "", src: "d.com", id: "d", name: "d", blocked: null, reasons: [], children: []}
+  ]
+  id: "",
+  name: "",
+  reasons: [],
+  src: "",
+  url:"a.com"
+}
+```
+
+See the [Security and privacy](https://github.com/rubberyuzu/bfcache-not-retored-reason/blob/main/NotRestoredReason.md#security-and-privacy) section in the explainer for more details about security and privacy considerations.
+
+## Blocking reasons
+
+As we said earlier, there are many different reasons why blocking could occur. We have compiled a [handy spreadsheet](https://docs.google.com/spreadsheets/d/1li0po_ETJAIybpaSX5rW_lUN62upQhY0tH4pR5UPt60/edit#gid=0) showing all the reason strings and explaining what they mean.
+
+There are a few major categories of reason that are worth calling out:
+
+* Circumstantial: this refers to blocking reasons not directly related to the developer's page code. For example a related component crashed, something went wrong with the loading process, the page is in a temporary state that can't be cached, BFCache disabled due to insufficient memory, or a service worker did something to the page that disqualifies it from being cached. 
+* Extensions: there are a few different reason messages related to extensions. Generally we combine quite a few different reasons into the "Extensions" reason. We are intentionally vague about extension-related blocking reasons because we don't want to give away too much information about what extensions the user have installed, which ones are active on the page, what they are doing, etc.
+* PageSupportNeeded: The developer's code is using a web platform feature that is otherwise not BFCache blocking, but it is currently in a state that is BFCache blocking. For example, the page currently has a [BroadcastChannel](https://developer.mozilla.org/docs/Web/API/BroadcastChannel) with registered listeners, or an open [IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) connection.
+* SupportPending: The developer's code is using a web platform feature that disqualifies the page from the BFCache, for exaple the [Web Serial API](https://developer.mozilla.org/docs/Web/API/Web_Serial_API), [Web Authentication API](https://developer.mozilla.org/docs/Web/API/Web_Authentication_API), [File System Access API](https://developer.mozilla.org/docs/Web/API/File_System_Access_API), or [Media Session API](https://developer.mozilla.org/docs/Web/API/Media_Session_API). This category is also used to report the presence of a tool outside the page iteself that is blocking BFCaching, such as a screenreader or the Chrome password manager.
```

Comment:
Added two of the most common reasons:

```suggestion
* `Circumstantial`: this refers to blocking reasons not directly related to the developer's page code. For example a related component crashed, something went wrong with the loading process, the page is in a temporary state that can't be cached, bfcache disabled due to insufficient memory, or a service worker did something to the page that disqualifies it from being cached. 
* `Extensions`: there are a few different reason messages related to extensions. Generally we combine quite a few different reasons into the "Extensions" reason. We are intentionally vague about extension-related blocking reasons because we don't want to give away too much information about what extensions the user have installed, which ones are active on the page, what they are doing, etc.
* `PageSupportNeeded`: The developer's code is using a web platform feature that is otherwise not BFCache blocking, but it is currently in a state that is bfcache blocking. For example, the page currently has a [BroadcastChannel](https://developer.mozilla.org/docs/Web/API/BroadcastChannel) with registered listeners, or an open [IndexedDB](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) connection.
* `Unload Handler`: The page has registered an unload handler, which currently [prevents the bfcache being used in some browsers](https://web.dev/bfcache/#never-use-the-unload-event).
* `MainResourceHasCacheControlNoStore`: The page is using `Cache-Control: no-store`, which currently [prevents the bfcache being used in some browsers](https://web.dev/bfcache/#minimize-use-of-cache-control-no-store).
* `SupportPending`: The developer's code is using a web platform feature that disqualifies the page from the bfcache, for example the [Web Serial API](https://developer.mozilla.org/docs/Web/API/Web_Serial_API), [Web Authentication API](https://developer.mozilla.org/docs/Web/API/Web_Authentication_API), [File System Access API](https://developer.mozilla.org/docs/Web/API/File_System_Access_API), or [Media Session API](https://developer.mozilla.org/docs/Web/API/Media_Session_API). This category is also used to report the presence of a tool outside the page itself that is blocking the bfcache, such as a screenreader or the Chrome password manager.
```